src/Utilities/SidreDataCollectionInline.hh

```diff
@@ -0,0 +1,214 @@
+   int view_count = 0;
+
+   for (u_int i = 0; i < field.size(); ++i)
+   {
+      axom::IndexType num_elements = field[i].size();
+      const DataType *data = &(*field[i].begin());
+      wholeField->createView(view_name + std::to_string(view_count), dtype, num_elements, (void*)data);
+      view_count++;
+   }
+   return wholeField;
+}
+
+//------------------------------------------------------------------------------
+template<typename Dimension, typename DataType>
+inline
+axom::sidre::Group *SidreDataCollection::sidreStoreField(const std::string &view_name,
+                                                         const Spheral::Field<Dimension, std::tuple<DataType, DataType, DataType>> &field)
+{
+   axom::sidre::DataTypeId dtype = field.getAxomTypeID();
+   axom::sidre::Group* wholeField = m_datastore_ptr->getRoot()->createGroup(view_name + "_Group");
+   axom::IndexType num_elements = 3;
+   int view_count = 0;
+
+   for (u_int i = 0; i < field.size(); ++i)
+   {
+      DataType data [] = {std::get<0>(field[i]), std::get<1>(field[i]), std::get<2>(field[i])};
+      axom::sidre::Buffer* buff = m_datastore_ptr->createBuffer()->allocate(dtype, num_elements)
+                                                 ->copyBytesIntoBuffer(data, sizeof(DataType) * num_elements);
+      wholeField->createView(view_name + std::to_string(view_count), dtype, num_elements, buff);
+      view_count++;
+   }
+   return wholeField;
+}
+
+//------------------------------------------------------------------------------
+template<typename Dimension, typename DataType>
+inline
+axom::sidre::Group *SidreDataCollection::sidreStoreField(const std::string &view_name,
+                                                         const Spheral::Field<Dimension, std::tuple<DataType, DataType, DataType, DataType>> &field)
+{
+   axom::sidre::DataTypeId dtype = field.getAxomTypeID();
+   axom::sidre::Group* wholeField = m_datastore_ptr->getRoot()->createGroup(view_name + "_Group");
+   axom::IndexType num_elements = 4;
+   int view_count = 0;
+
+   for (u_int i = 0; i < field.size(); ++i)
+   {
+      DataType data [] = {std::get<0>(field[i]), std::get<1>(field[i]), std::get<2>(field[i]), std::get<3>(field[i])};
+      axom::sidre::Buffer* buff = m_datastore_ptr->createBuffer()->allocate(dtype, num_elements)
+                                                 ->copyBytesIntoBuffer(data, sizeof(DataType) * num_elements);
+      wholeField->createView(view_name + std::to_string(view_count), dtype, num_elements, buff);
+      view_count++;
+   }
+   return wholeField;
+}
+
+//------------------------------------------------------------------------------
+template<typename Dimension, typename DataType>
+inline
+axom::sidre::Group *SidreDataCollection::sidreStoreField(const std::string &view_name,
+                                                         const Spheral::Field<Dimension, std::tuple<DataType, DataType, DataType, DataType, DataType>> &field)
```

Comment:
The parameters are lined up, but the second one is very long. Do I need break it up onto more lines?

Reply:
It's not the prettiest, if you can break it up into a format that maintains the readability go for it, but I don't think it's too much of an issue, this only really effects the tuple cases.